draft-ietf-mls-protocol.md

```diff
@@ -1465,69 +1467,59 @@ The ciphertext field of the MLSCiphertext object is produced by
-
-The members learn the relevant path secrets by decrypting one of the encrypted
-path secrets in the DirectPath and working back to the root (as in normal
-DirectPath processing).
-
-Note the degenerate case of this process: The group creator can initialize a
-group with one member (itself) and add the remaining members by sending Add
-proposals followed by a Commit.  This approach is slightly more verbose than
-simply sending an Init message, but also slightly more private.  The Init
-leaks the initial configuration of the group, while in the Add approach, the
-initial group state is sent in encrypted Welcome messages.
-
-[[ OPEN ISSUE: It might be better to collapse these two approaches.  This would
-entail sending multiple Proposals alongside a Commit and encrypting a Welcome to
-multiple participants, both of which seen like nice optimizations anyway. ]]
-
-[[ OPEN ISSUE: It might be desireable for the group creator to be
-able to "pre-warm" the tree, by providing values for some nodes not
-on its direct path.  This would violate the tree invariant, so we
-would need to figure out what mitigations would be necessary. ]]
+The creator of a group MUST take the following steps to initialize the group:
+
+* Fetch ClientInitKeys for the members to be added, and selects a version and
+  ciphersuite according to the capabilities of the members. [[ TODO: Discuss
+  downgrade prevention here ]]
+
+* Initialize a one-member group with the following initial values (where "0"
+  represents an all-zero vector of size Hash.length):
+  * Ratchet tree: A tree with a single node, a leaf containing an HPKE public
+    key and credential for the creator
+  * Group ID: A value set by the creator
+  * Epoch: 0x00000000
+  * Tree hash: The root hash of the above ratchet tree
+  * Confirmed transcript hash: 0
+  * Interim transcript hash: 0
+  * Init secret: 0
+
+* For each member, construct an Add proposal from the ClientInitKey for that
+  member (see {{add}})
+
+* Construct a Commit message that commits all of the Add proposals, in any order
+  chosen by the creator (see {{commit}})
+
+* Process the Commit message to obtain a new group state (for the epoch in which
+  the new members are added) and a Welcome message
+
+* Transmit the Welcome message to the other new members
+
+The recipient of a Welcome message processes it as described in
+{{welcoming-new-members}}.
+
+In principle, the above process could be streamlined by having the creator
+directly create a tree and choose a random value for first epoch's epoch secret.
+We follow the steps above because it removes unnecessary choices, by which, for
+example, bad randomness could be introduced.  The only choices the creator makes
+here are its own HPKE key and credential, and the leaf secret from which the
+Commit is built.
+
+A new member receiving a Welcome message can recognize group creation if the
+signer of the Welcome message is at index 0, and the number of entries in the
```

Comment:
I would prefer avoiding the requirement that the creator is at index 0 because it breaks privacy of its identity.
When newcomers are included later on, it might be nice to not know immediately who is the creator.